plugins/plugin-site/src/components/Plugin.jsx

```diff
@@ -25,6 +25,33 @@ function Developers({developers}) {
 Developers.propTypes = PluginDevelopers.propTypes;
 
 function Plugin({plugin: {name, title, stats, labels, excerpt, developers, buildDate, releaseTimestamp}}) {
+    let progress = 0;
+    let color =
+    progress > 80 ? "#87d068" : progress > 60 ? "#e5e512" : "#fe3535";
+
+    const graphqlData = useStaticQuery(graphql`
+        query {
```

Comment:
can you pull this data on the pages instead of the components, then you can filter the data for the specific plugin.
useStaticQuery generates a json file that gets loaded into memory, so that will grow really fast.

Reply:
I have to pull those data in [link](https://github.com/Harsh3341/plugin-site/blob/master/plugins/plugin-site/src/templates/search.jsx) am I right?

Reply:
I guess depends where you want it. 

I havn't looked at your preview site yet, just skimmed the code while working.

Each src/template maps to a page, see https://github.com/jenkins-infra/plugin-site/blob/master/plugins/plugin-site/gatsby-node.js#L25-L54

each src/template should have a pageQuery which loads json data for that specific page.

Reply:
oh this is only shown on the search page? I think that data should come from the algoila search then, since thats doing the filtering and pagination not graphql.

Reply:
@Harsh3341 you can check https://github.com/jenkins-infra/plugin-site/blob/master/plugins/plugin-site/src/utils/algolia-queries.js which defines what data goes to Algolia

Reply:
In the new commit, I have pulled pluginHealthScore data in [link](https://github.com/Harsh3341/plugin-site/blob/displaying-plugin-health-score/plugins/plugin-site/src/templates/search.jsx)